src/Model/BaseBlock.php

```diff
@@ -44,7 +46,9 @@ abstract class BaseBlock implements BlockInterface
     protected $parent;
 
     /**
-     * @var BlockInterface[]
+     * NEXT_MAJOR: Restrict typehint to Collection.
+     *
+     * @var Collection<int, BlockInterface>|array<BlockInterface>
```

Comment:
what is the advantage over iterable?

Reply:
We're using `count`: https://github.com/sonata-project/SonataBlockBundle/blob/4.x/src/Model/BaseBlock.php#L224

Reply:
so `array<BlockInterface>` would be fine then? :thinking: